serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+What is the best way to use these pages? 
+The answer is that every problem has different characteristics and caveats, and you should be able to understand _if you need_ to learn a new tool, or you can get the things done with those that you already know! 
+So, use these pages to compare the pros and cons of the hundreds of tools that exist!
+
+- [Awesome AI-ML-DL](https://github.com/neomatrix369/awesome-ai-ml-dl)
+- [Awesome Machine Learning](https://github.com/josephmisiti/awesome-machine-learning)
+- [Amazing ML OpenSource 2019](https://github.com/Mybridge/amazing-machine-learning-opensource-2019)
+- [Awesome Machine Learning Model Management Frameworks](https://github.com/awoerner92/awesome-ml-tools)
+- [A Curated List of Awesome Machine Learning Frameworks, Libraries and Softwares](https://www.mo-data.com/a-curated-list-of-awesome-machine-learning-frameworks-libraries-and-software/)
+
+
+# Production Ready Tools
+In this section, Virgilio listed for you some "Production-Ready" tools that can speed up the process of creating value with data. You can use these tools along with those listed in the previous sections.
+
+For a comprehensive list of production-ready tools, be sure to take a tour at:
+
+- [Awesome Production Machine Learning](https://github.com/EthicalML/awesome-production-machine-learning#data-labelling-tools-and-frameworks)
+
+Virgilio has chosen for you the best tools of this type for 3 problem categories:
+
+## Computer Vision
+
+- [OpenCV](https://opencv.org/about/)
+
+OpenCV (Open Source Computer Vision Library) is an open-source computer vision and machine learning software library. OpenCV was built to provide a common infrastructure for computer vision applications and to accelerate the use of machine perception in commercial products. Being a BSD-licensed product, OpenCV makes it easy for businesses to utilize and modify the code.
+
+Learn how to use OpenCV here:
+
+- [OpenCV Python for Beginners - Full Course in 10 Hours (2020)](https://www.youtube.com/watch?v=N81PCpADwKQ)
+
+Check also Virgilio's tutorials in the Inferno section, They all use OpenCV!
+
+## Natural Language Processing
+
+- [Spacy](https://spacy.io/)
+
+SpaCy is a free, open-source library for advanced Natural Language Processing (NLP) in Python.
+
+If you’re working with a lot of text, you’ll eventually want to know more about it. For example, what’s it about? What do the words mean in context? Who is doing what to whom? What companies and products are mentioned? Which texts are similar to each other?
+
+spaCy is designed specifically for production use and helps you build applications that process and “understand” large volumes of text. It can be used to build information extraction or natural language understanding systems or to pre-process text for deep learning.
+
+They offer a free course on their website:
+
+- [Advanced NLP with Spacy](https://course.spacy.io/)
+
```

Comment:
I can recommend this library: https://bit.ly/better-nlp-launch, plenty of resources and mostly high-level and also notebooks to learn from, zero-install out-of-the-box containers.

Reply:
perfect very good I add it.

Reply:
Yes but if people does not know how to use containers?

Reply:
They don;t need to use containers, they can load the notebooks in Kaggle or Google Colab, the repo allows different targets:
- local machine
- docker container
- google colab
- kaggle kernel

BUt what if they don't know how to use notebooks or kaggle kernels - that is also a thing right ;)

Reply:
I'd prefer to leave them play with "first-order" librarie, before giving them an higher level library.. maybe we can add it inside a "advanced nlp guide" or smth like that